R/generic-methods.R

```diff
@@ -125,4 +125,21 @@ interpolate_dfs <- function(x, from, to, ...) UseMethod("interpolate_dfs")
 
 #' @export
 #' @rdname interpolate_dfs
-interpolate_fwds <- function(x, from, to, ...) UseMethod("interpolate_fwds")
\ No newline at end of file
+interpolate_fwds <- function(x, from, to, ...) UseMethod("interpolate_fwds")
+
+
+
+#' Interpolate volatility surface
+#'
+#' This interpolates a vol surface from the class [VolSurface] at a date and a strike
+#' @param x the object to interpolate
+#' @param maturity a vetor of type date with the dates to interpolate at
+#' @param strike a vector of stikes to interpolate at with the same length as `date` vector
+#' @param ... other parameters used in the interpolation
+#' @return Returns a numeric vector of the same length as `date` parameter containing the interpolated
+#' implied volatilities at each couple of `(date, strike)`.
+#' @export
+#' @aliases interpolate_vol
+interpolate_vol <- function(x, maturity, strike, ...) UseMethod("interpolate_vol")
```

Comment:
Why can't you extend `interpolate()` by implementing a specific method for `VolSurface`?

Reply:
Good idea ! It is now inherited by VolSurface class. I previously thought to follow the same logic as interpolate_dfs, interpolate_zeros, interpolate_fwd; but this is actually different.